bot/cogs/base_converter_cog.py

```diff
@@ -0,0 +1,129 @@
+        o = oct(number)	
+        h = hex(number)
+        if number in range(1114112): 
+            a = chr(number)
+        else:
+            a = 'Unicode representation not found.'
+
+        embed = discord.Embed(title='Conversions', description =f'Numerical Conversions of {str(number)}', color = Colors.ClemsonOrange)
+        embed.add_field(name='Decimal', value=d, inline=False)
+        embed.add_field(name='Hexadecimal', value=h, inline=False)
+        embed.add_field(name='Octal', value=o, inline=False)
+        embed.add_field(name="UTF-8", value=a, inline=False)
+        
+        await ctx.send(embed=embed)
+        
+    @bin.error
+    async def bin_error(self, ctx, error):
+        if isinstance(error, commands.BadArgument):
+            await ctx.send('Input Binary is Invalid')
+        if isinstance(error, commands.CommandInvokeError):
+            await ctx.send('Input Binary is Invalid')
+
+    @baseconvert.command()
+    async def dec(self, ctx, number : int):
+        b = bin(number)
+        h = hex(number)
+        o = oct(number)
+        if number in range(1114112):
+            a = chr(number)
+        else:
+            a = 'Unicode representation not found.'
+
+        embed = discord.Embed(title='Conversions', description =f'Numerical Conversions of {str(number)}', color = Colors.ClemsonOrange)
+        embed.add_field(name='Binary', value=b, inline=False)
+        embed.add_field(name='Hexadecimal', value=h, inline=False)
+        embed.add_field(name='Octal', value=o, inline=False)
+        embed.add_field(name='UTF-8', value=a, inline=False)
+        
+        await ctx.send(embed=embed)
+
+    @dec.error
+    async def dec_error(self, ctx, error):
+        if isinstance(error, commands.BadArgument):
+            await ctx.send('Input Decimal is Invalid')
+        if isinstance(error, commands.CommandInvokeError):
+            await ctx.send('Input Decimal is Invalid')
+
+    @baseconvert.command()
+    async def hex(self, ctx, number : str):
+        number = int(number, 16)
+        b = bin(number)
+        d = int(number)
+        o = oct(number)
+
+        if number in range(1114112):
+            a = chr(number)
+        else:
+            a = 'Unicode representation not found.'
+
+        embed = discord.Embed(title='Conversions', description =f'Numerical Conversions of {str(number)}', color = Colors.ClemsonOrange)
+        embed.add_field(name='Binary', value=b, inline=False)
+        embed.add_field(name='Decimal', value=d, inline=False)
+        embed.add_field(name='Octal', value=o, inline=False)
+        embed.add_field(name='UTF-8', value=a, inline=False)
+        
+        await ctx.send(embed=embed)
+
+    @hex.error
+    async def hex_error(self, ctx, error):
+        if isinstance(error, commands.BadArgument):
+            await ctx.send('Input Hexadecimal is Invalid')
+        if isinstance(error, commands.CommandInvokeError):
+            await ctx.send('Input Hexadecimal is Invalid')
```

Comment:
We shouldnt handle errors here, we should preprocess the strings and test if they are valid and either allow the exception to bubble up or respond with an error embed, see some of the other larger cogs for an example of early exit embed conditions